### PR TITLE
Add 'require' argument to injectors

### PR DESCRIPTION
**Pull request by Aaron1011**
SpongeCommon | [SpongeForge](https://github.com/SpongePowered/SpongeForge/pull/491) | [SpongeVanilla](https://github.com/SpongePowered/SpongeVanilla/pull/203)

Now that we're using Mixin 0.4.10, a new parameter `require` is available in the various injectors (`@Inject`, `@Redirect`, `@ModifyArg`). This requires the injector to find at least as many targets as are specified by `require`.

By enabling this, we get an error when an injection fails, instead of it silently failing. While the `require` parameter can be leveraged to provide soft-failure (like `@Surrogate` for local var capture), all of our current injectors are required. This allows us to easily discover if a coremod is preventing an injection from working, and add another injection point to fix it.

### CODE REVIEW

**Comment by Aaron1011** [comment]
Pinging @SpongePowered/developers for review.

File: src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenBigMushroom.java [hunk 1]
```diff
@@ -67,7 +67,7 @@
     private Function<Location<Chunk>, PopulatorObject> override = null;
     private VariableAmount mushroomsPerChunk;
 
-    @Inject(method = "<init>", at = @At("RETURN") )
+    @Inject(method = "<init>", at = @At("RETURN") , require = 1)
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenBigMushroom.java]: Formatting is off

File: src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenBlockBlob.java [hunk 2]
```diff
@@ -54,7 +54,7 @@
     private VariableAmount radius;
     private VariableAmount count;
 
-    @Inject(method = "<init>(Lnet/minecraft/block/Block;I)V", at = @At("RETURN") )
+    @Inject(method = "<init>(Lnet/minecraft/block/Block;I)V", at = @At("RETURN") , require = 1)
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenBlockBlob.java]: Formatting is off

File: src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenCactus.java [hunk 3]
```diff
@@ -49,7 +49,7 @@
     private VariableAmount cactiPerChunk;
     private VariableAmount height;
 
-    @Inject(method = "<init>()V", at = @At("RETURN") )
+    @Inject(method = "<init>()V", at = @At("RETURN") , require = 1)
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenCactus.java]: Formatting is off

File: src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenDeadBush.java [hunk 4]
```diff
@@ -47,7 +47,7 @@
 
     private VariableAmount count;
 
-    @Inject(method = "<init>", at = @At("RETURN") )
+    @Inject(method = "<init>", at = @At("RETURN") , require = 1)
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/world/gen/populators/MixinWorldGenDeadBush.java]: Formatting is off

**Comment by kashike** [comment]
Formatting is off on a lot of injections

**Pull request closed without merging**